### PR TITLE
ui bug fixes

### PR DESCRIPTION
**Pull request by shubham-shay-te**

### CODE REVIEW

File: website/src/fieldsGenerator/dataAccessFields.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ export const dataAccessFields = (path, yamlData) => {
     const connections = yamlData.core_lib.connections
     const entities = yamlData.core_lib.entities
     const fields = []
-    const connection = []
-    const entity = []
+    const connection = ['']
```
Comment (shay-te) [inline website/src/fieldsGenerator/dataAccessFields.js:11]: what does this solve?

File: website/src/fieldsGenerator/serviceFields.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { getBoolean } from '../utils/commonUtils';
 
 export const serviceFields = (path, yamlData) => {
     const fields = []
-    const dataAccess = []
+    const dataAccess = ['']
```
Comment (shay-te) [inline website/src/fieldsGenerator/serviceFields.js:5]: what is the reason that an empty string is needed?

File: website/src/utils/YamlData.js [hunk 3]
```diff
@@ -102,10 +103,21 @@ export class YamlData {
     }
 
     delete(path) {
-        const data = JSON.parse(JSON.stringify(this.yaml))
+        let data = JSON.parse(JSON.stringify(this.yaml))
         const steps = path.split(".")
         const parent = getValueAtPath(data, steps.slice(0, -1));
-        parent.splice(steps.at(-1), 1);
+        if(path.includes('connections')){
```
Comment (shay-te) [inline website/src/utils/YamlData.js:109]: this code is specific to a connection. and it is inside the YamlData class. 
Haven't we said that specific handling is else where?

File: website/src/fieldsGenerator/dataAccessFields.js (hunk 1 above)
Comment (shubham-shay-te) [reply website/src/fieldsGenerator/dataAccessFields.js:11]: this is for the user to select a empty connection

File: website/src/fieldsGenerator/serviceFields.js (hunk 2 above)
Comment (shubham-shay-te) [reply website/src/fieldsGenerator/serviceFields.js:5]: for the user to select empty dataaccess

**Review by shay-te — APPROVED** [review]

**Pull request merged**